src/Core/Result.php

```diff
@@ -74,6 +61,35 @@ protected function populateFromResponse(ResponseInterface $response): void
         if (300 <= $statusCode) {
             throw new RedirectionException($this->response);
         }
+
+        $this->populateFromResponse($this->response);
```

Comment:
Here is the issue. As soon as we run resolve, we execute this function. 

This function starts to take stuff from the response. 

* `$response->getStatusCode()` - Download only the first line of the request. 
* `$response->getHeaders()` - Download the headers
* `$response->getContent()` - Download the body

So if you never call `$response->getContent()` you never download the body. But we call `$response->getHeaders()` and `$response->getContent()` as soon as we resolve the Result.